e2e/tables.spec.ts

```diff
@@ -0,0 +1,430 @@
+      },
+      {
+        item: 'socks',
+        quantity: 475,
+        stock_date: '2022-10-02',
+        color: 'green',
+        search_terms: ['wool', 'machine wash'],
+      },
+    ]
+
+    const updatedColumnsSchemaConfig = [
+      ...initialColumnsSchemaConfig,
+      { name: 'reorder_quantity', type: 'Integer', defaultValue: '25' },
+    ]
+    const updatedTableData = initialTableData.map((data, index) => ({
+      ...data,
+      reorder_quantity: '25',
+    }))
+
+    await test.step('User goes to tables page', async () => {
+      await goToDashboardPage(userPage, getTablesPath(userProject.id))
+      await expectTablesPageLoaded(userPage, userProject.id)
+      await expect(userPage.getByText(noTablesText)).toBeVisible()
+    })
+
+    // TODO: remove this step after react-based table column schema editor is
+    // released from experimental mode
+    await test.step('Toggle into Experimental Mode to use react-based table column schema editor', async () => {
+      await toggleIntoExperimentalMode(userPage)
+      await expectTablesPageLoaded(userPage, userProject.id)
+      await expect(userPage.getByText(noTablesText)).toBeVisible()
+    })
+
+    await test.step('User creates a table', async () => {
+      const createTableHeader =
+        await test.step('open table creation modal', async () => {
+          await userPage.getByRole('button', { name: 'Add New...' }).click()
+          await userPage.getByRole('menuitem', { name: 'Add Table' }).click()
+          const createTableHeader = userPage.getByRole('heading', {
+            name: 'Create Table',
+          })
+          await expect(createTableHeader).toBeVisible()
+          return createTableHeader
+        })
+
+      await test.step('enter table name and description', async () => {
+        await userPage
+          .locator('div')
+          .filter({ hasText: /^Name$/ })
+          .getByRole('textbox')
+          .fill(tableName)
+        await userPage
+          .locator('div')
+          .filter({ hasText: /^Description$/ })
+          .locator('textarea')
+          .fill(tableDescription)
+        await userPage.getByRole('button', { name: 'Next' }).click()
+      })
+
+      await test.step('create table schema', async () => {
+        await addColumnsFromColumnSchemaConfig(
+          userPage,
+          initialColumnsSchemaConfig,
+        )
+      })
+
+      await test.step('finish creating table', async () => {
+        await userPage.getByRole('button', { name: 'Finish' }).click()
+        await expect(
+          userPage.getByText('Updating the table schema...'),
+        ).toBeVisible()
+        await expect(createTableHeader).not.toBeVisible()
+      })
+    })
+
+    await test.step('Table has been created with expected schema', async () => {
+      await expectTablePageLoaded(userPage, tableName, tableDescription)
+
+      await test.step('get table entity id', async () => {
+        tableEntityId =
```

Comment:
Table entity will be moved to the Trash Can in the test, but deleted via API in the `afterAll`